data/state/accountsDB.go

```diff
@@ -584,15 +636,42 @@ func (adb *AccountsDB) CancelPrune(rootHash []byte, identifier data.TriePruningI
 // SnapshotState triggers the snapshotting process of the state trie
 func (adb *AccountsDB) SnapshotState(rootHash []byte) {
 	log.Trace("accountsDB.SnapshotState", "root hash", rootHash)
-
+	adb.mainTrie.EnterSnapshotMode()
 	adb.mainTrie.TakeSnapshot(rootHash)
+
+	go func() {
```

Comment:
I would eliminate this go routine. The peer account DB is done in sysc.

Reply:
Go routine is a must here as the process may take a long time. The accountsDB sets storage manager to snapshot mode, so there is no more prunning. After that it calculates what needs to be snapshotted and snapshot is called. The blockprocessor should not stay and wait for the whole process to be finished. Even in case of peerAccountsDB the trieStorageManager is announced to do the snapshotting, but the actual processing is in go routine.